draft-ietf-httpbis-sfbis.md

```diff
@@ -689,14 +689,18 @@ Given a Date as input_date, return an ASCII string suitable for use in an HTTP f
 
 Given a string of Unicode characters as input_string, return an ASCII string suitable for use in an HTTP field value.
 
-0. Let byte_array be the result of applying UTF-8 encoding {{UTF8}} to input_string. If there is an error in doing so, fail parsing.
-1. Let encoded_string be an empty string.
-2. For each byte in byte_array:
+0. If input_string is not a string of Unicode characters, fail parsing.
+1. Let byte_array be the result of applying UTF-8 encoding {{UTF8}} to input_string.
+2. Let encoded_string be an empty string.
+3. For each byte in byte_array:
    1. If byte is %x25 ("%"), append "%25" to encoded_string.
+   2. If byte is %x22 (DQUOTE), append "%22" to encoded_string.
    2. If byte is in the ranges %x00-1f or %x7f-ff, apply the percent-encoding defined in {{Section 2.1 of URI}} to byte and append the result to encoded_string.
```

Comment:
Aside from the counting strangeness (2x2), perhaps it would be better if you spelled this out rather than doing it by reference.  That is, append % (0x25).  Encode the byte value as a two character, uppercase hexadecimal string {{cite Section 8 of RFC 4648 if you like}} and append the UTF-8 encoding of those characters.

(Ulterior motive: I would much prefer that we avoid relying on RFC 3986 for this.)

Reply:
If by "2x2" you mean the fact that there are two "2." items, markdown will sort it out.

I _think_ what you're suggesting is roughly:

n. If byte is in the range %x00-1f, %x22, %x25, or in the range %x7f-ff:
    1. Append "%" to encoded_string.
    2. Append the percent encoding to byte...

Is that roughly what you have in mind?

Re: 4648 vs 3986 -- I strongly suspect that people will be reaching for their URI encoding libraries for this, not their Base16 implementations. That's one of the reasons I'm a bit nervous about specifying case -- it's not sensitive in URLs.